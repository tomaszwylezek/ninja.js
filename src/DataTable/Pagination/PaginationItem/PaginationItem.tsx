import React, { FC, SyntheticEvent } from 'react';
import cn from 'classnames';

export interface IPaginationItemProps {
  pageNumber: number;
  isSelected: boolean;
  onChange: (event: SyntheticEvent<HTMLButtonElement>) => void;
}

export const PaginationItem: FC<IPaginationItemProps> = ({ pageNumber, isSelected, onChange }) => (
  <li className="page-item mr-1">
    <button
      className={cn('page-link', {
        'button-outline': isSelected
      })}
      onClick={onChange}
    >
      {pageNumber + 1}
    </button>
  </li>
);
