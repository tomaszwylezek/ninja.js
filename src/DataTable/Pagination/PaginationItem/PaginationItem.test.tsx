import React from 'react';
import { mount, shallow } from 'enzyme';
import { PaginationItem, IPaginationItemProps } from './PaginationItem';

describe('PaginationItem', () => {
  let props: IPaginationItemProps;

  beforeEach(() => {
    props = {
      isSelected: false,
      onChange: jest.fn(),
      pageNumber: 0
    };
  });

  it('renders without crashing', () => {
    const component = shallow(<PaginationItem {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('calls a function when button clicked', () => {
    const component = mount(<PaginationItem {...props} />);

    component.find('button').simulate('click');

    expect(props.onChange).toHaveBeenCalled();
  });

  it('add a class to button when is current page', () => {
    const component = mount(<PaginationItem {...props} />);

    component.setProps({
      isSelected: true
    });

    expect(component.find('button').hasClass('button-outline')).toBe(true);
  });
});
