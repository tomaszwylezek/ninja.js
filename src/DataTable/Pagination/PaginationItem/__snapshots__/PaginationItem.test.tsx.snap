// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem renders without crashing 1`] = `
<li
  className="page-item mr-1"
>
  <button
    className="page-link"
    onClick={[MockFunction]}
  >
    1
  </button>
</li>
`;
